docs/tutorials/host_apt_news.rst

```diff
@@ -0,0 +1,225 @@
+accomplish this with by installing and configuring `nginx`_ to serve a
+properly formatted JSON file.
+
+First, enter the ``apt-news-server`` container:
+
+.. code-block:: console
+
+   ubuntu@apt-news-tutorial:~$ lxc shell apt-news-server
+   root@apt-news-server:~#
+
+In ``apt-news-server``, install ``nginx`` via ``apt``. This will also start the HTTP server on port 80.
+
+.. code-block:: console
+
+   root@apt-news-server:~# apt install nginx
+
+With ``nginx`` installed and running, we can author an APT news JSON file. Open ``/var/www/html/aptnews.json``:
+
+.. code-block:: console
+
+   root@apt-news-server:~# nano /var/www/html/aptnews.json
+
+and add the following content:
+
+.. code-block:: json
+
+   {
+     "messages": [
+       {
+         "begin": "TODAY",
+         "selectors": {
+           "codenames": ["jammy"]
+         },
+         "lines": [
+           "Hello 22.04 users!",
+           "This is the APT news server."
+         ]
+       },
+       {
+         "begin": "TODAY",
+         "lines": [
+           "Hello everyone else!",
+           "This is the APT news server."
+         ]
+       }
+     ]
+   }
+
+That apt news configuration will show one message to systems running Ubuntu
+22.04 (Codename "jammy") and will show another message to all other systems.
+
+The value of ``"begin"`` actually needs to be an ISO8601 formatted datetime
+string, and the message won't be shown before the ``begin`` date or more than 30
+days after the ``begin`` date. For the purposes of this tutorial you can
+quickly replace "TODAY" with today's date in ``aptnews.json`` with the
+following command.
+
+.. code-block:: console
+
+   root@apt-news-server:~# sed -i "s/TODAY/$(date --iso-8601=seconds)/" /var/www/html/aptnews.json
+
+We're done configuring ``apt-news-server`` for now, so you can exit that container
+
+.. code-block:: console
+
+   root@apt-news-server:~# exit
+   ubuntu@apt-news-tutorial:~$
+
+
+Configure the client machines to use the APT news server
+========================================================
+
+We need to configure both client machines to use the APT news server we just
+created. This is a single command on each container that we can run using
+``lxc exec``:
+
+.. code-block:: console
+
+   ubuntu@apt-news-tutorial:~$ lxc exec client-noble -- pro config set apt_news_url=http://apt-news-server/aptnews.json
+   ubuntu@apt-news-tutorial:~$ lxc exec client-jammy -- pro config set apt_news_url=http://apt-news-server/aptnews.json
+
+That's it! Now those containers will start displaying APT news in the output of ``apt upgrade``.
+
+Normally, the containers would fetch the latest APT news whenever an
+``apt update`` is run, but at most once per day. For this tutorial, we'll use
+``pro refresh messages`` to force the containers to fetch the latest news
+right away.
+
+In the Jammy container we expect to see the special message for systems on
+Ubuntu 22.04.
+
+.. code-block:: console
```

Comment:
split this up into 4 code blocks
vm command
container commands
output
exit container

same for next one